test/common/formatter/substitution_formatter_test.cc

```diff
@@ -3327,17 +3329,17 @@ TEST(SubstitutionFormatterTest, StructFormatterOmitEmptyTest) {
 
   ProtobufWkt::Struct key_mapping;
   TestUtility::loadFromYaml(R"EOF(
-      test_key_filter_state: '%FILTER_STATE(nonexistent_key)%'
-      test_key_req: '%REQ(nonexistent_key)%'
-      test_key_res: '%RESP(nonexistent_key)%'
-      test_key_dynamic_metadata: '%DYNAMIC_METADATA(nonexistent_key)%'
+      test_key_root:
+        test_key_filter_state: '%FILTER_STATE(nonexistent_key)%'
+        test_key_req: '%REQ(nonexistent_key)%'
+        test_key_res: '%RESP(nonexistent_key)%'
+        test_key_dynamic_metadata: '%DYNAMIC_METADATA(nonexistent_key)%'
     )EOF",
                             key_mapping);
   StructFormatter formatter(key_mapping, false, true);
 
-  verifyStructOutput(
```

Comment:
Was it necessary to introduce a new function `verifyEmptyStructOutput`? Just curious.

Reply:
Yes, previously the test was flawed. `verifyStructOutput` did not check that only the expected keys were present (any unexpected keys must be absent).

I have now improved the function `verifyStructOutput`, so `verifyEmptyStructOutput` is not necessary anymore.